src/sonic-frr/patch/0054-support-vrrp6-commands-and-tracking-interface.patch

```diff
@@ -0,0 +1,3091 @@
++}
++
++/*
++ * XPath:
++ * /frr-interface:lib/interface/frr-vrrpd:vrrp/vrrp6-group/advertisement-interval
++ */
++DEFPY_YANG(no_vrrp6_advertisement_interval,
++      no_vrrp6_advertisement_interval_cmd,
++      "no vrrp6 (1-255)$vrid advertisement-interval [(10-40950)]",
++      NO_STR VRRP6_STR VRRP6_VRID_STR VRRP6_ADVINT_STR
++      "Advertisement interval in milliseconds; must be multiple of 10")
++{
++	nb_cli_enqueue_change(vty, "./advertisement-interval", NB_OP_MODIFY,
++			      NULL);
++
++	return nb_cli_apply_changes(vty, VRRP6_XPATH_ENTRY, vrid, VRRP6_FAMILY);
++}
++
++void cli_show_advertisement_interval_vrrp6(struct vty *vty, const struct lyd_node *dnode,
++				     bool show_defaults)
++{
++	const char *vrid = yang_dnode_get_string(dnode, "../virtual-router-id");
++	uint16_t advint = yang_dnode_get_uint16(dnode, NULL);
++
++	vty_out(vty, " vrrp6 %s advertisement-interval %u\n", vrid,
++		advint * CS2MS);
++}
++
++/*
++ * XPath:
++ * /frr-interface:lib/interface/frr-vrrpd:vrrp/vrrp6-group/v6/virtual-address
++ */
++DEFPY_YANG(vrrp6_ip6,
++      vrrp6_ip6_cmd,
++      "[no] vrrp6 (1-255)$vrid ipv6 X:X::X:X",
++      NO_STR
++      VRRP6_STR
++      VRRP6_VRID_STR
++      "Add IPv6 address\n"
++      VRRP6_IP_STR)
++{
++	int op = no ? NB_OP_DESTROY : NB_OP_CREATE;
++	nb_cli_enqueue_change(vty, "./v6/virtual-address", op, ipv6_str);
++
++	return nb_cli_apply_changes(vty, VRRP6_XPATH_ENTRY, vrid, VRRP6_FAMILY);
++}
++
++void cli_show_ipv6_vrrp6(struct vty *vty, const struct lyd_node *dnode, bool show_defaults)
++{
++	const char *vrid =
++		yang_dnode_get_string(dnode, "../../virtual-router-id");
++	const char *ipv6 = yang_dnode_get_string(dnode, NULL);
++
++	vty_out(vty, " vrrp6 %s ipv6 %s\n", vrid, ipv6);
++}
++
++/*
++ * XPath:
++ * /frr-interface:lib/interface/frr-vrrpd:vrrp/vrrp6-group/track-interface
++ */
++DEFPY_YANG(vrrp6_track_interface,
++      vrrp6_track_interface_cmd,
++      "vrrp6 (1-255)$vrid track-interface IFNAME$intf_name priority-dec (10-50)",
++      VRRP6_STR
++      VRRP6_VRID_STR
++      "Add tracking interface\n"
++      VRRP6_TRACK_INTF_STR
++	  VRRP6_PRIORITY_DECREMENT_STR
++	  "Priority decrement value\n")
++{
++	char xpath[XPATH_MAXLEN];
++	char xpath_track[XPATH_MAXLEN + 32];
++
++	snprintf(xpath, sizeof(xpath),
++			"./track-interface[interface='%s']",
++			intf_name);
++	nb_cli_enqueue_change(vty, xpath, NB_OP_CREATE, NULL);
++
++	snprintf(xpath_track, sizeof(xpath_track),
++			"%s/priority-decrement", xpath);
++	nb_cli_enqueue_change(vty, xpath_track, NB_OP_MODIFY, priority_dec_str);
++
++	return nb_cli_apply_changes(vty, VRRP6_XPATH_ENTRY, vrid, VRRP6_FAMILY);
++}
++
++DEFPY_YANG(no_vrrp6_track_interface,
++      no_vrrp6_track_interface_cmd,
++      "no vrrp6 (1-255)$vrid track-interface IFNAME$intf_name",
++      NO_STR
++      VRRP6_STR
++      VRRP6_VRID_STR
++      "Add tracking interface\n"
++      VRRP6_TRACK_INTF_STR
++      VRRP6_PRIORITY_DECREMENT_STR
++      "Priority decrement value\n")
```

Comment:
VRRP6_PRIORITY_DECREMENT_STR need in this 'no' command?

Reply:
It will be removed, thanks.